### PR TITLE
Enable firefox integration tests

### PR DESCRIPTION
**Pull request by nturgut**
Enable firefox integration tests

LUCI run that shows the firefox integration tests are passing: https://chromium-swarm.appspot.com/task?id=4ce8d623bd36bb10

Note: I added the following cipd package flutter_internal/browser-drivers/firefoxdriver-linux in the recipe.

fixes: https://github.com/flutter/flutter/issues/59650

### CODE REVIEW

File: lib/web_ui/dev/driver_manager.dart [hunk 1]
```diff
@@ -60,6 +62,61 @@ class ChromeDriverManager extends DriverManager {
...
+    await startProcess('./firefoxdriver/geckodriver', ['--port=4444'],
+        workingDirectory: driverPath);
+    print('INFO: Driver started');
+  }
+
+  /// Get the geckodriver version to be used with [FirefoxDriverInstaller].
+  ///
+  /// For different versions of geckodriver. See:
+  /// https://github.com/mozilla/geckodriver/releases
+  static String getLockedDeckoDriverVersion() {
```
Comment (yjbanov) [inline lib/web_ui/dev/driver_manager.dart:113]: typo: s/Decko/Gecko/

File: lib/web_ui/dev/driver_manager.dart [hunk 2]
```diff
@@ -60,6 +62,61 @@ class ChromeDriverManager extends DriverManager {
...
+    io.Directory.current = _browserDriverDir;
+
+    try {
+      await firefoxDriverInstaller.install(alwaysInstall: false);
+    } finally {
+      io.Directory.current = temp;
+    }
+  }
+
+  /// Throw an error if driver directory does not exists.
```
Comment (yjbanov) [inline lib/web_ui/dev/driver_manager.dart:92]: nit: exist

File: lib/web_ui/dev/driver_manager.dart [hunk 3]
```diff
@@ -60,6 +62,61 @@ class ChromeDriverManager extends DriverManager {
...
+  ///
+  /// Driver should already exist on LUCI as a CIPD package.
+  Future<void> _verifyDriverForLUCI() {
+    if (!_browserDriverDir.existsSync()) {
+      throw StateError('Failed to locate Firefox driver on LUCI on path:'
+          '${_browserDriverDir.path}');
+    }
+    return Future<void>.value();
```
Comment (yjbanov) [inline lib/web_ui/dev/driver_manager.dart:100]: Why return asynchronously?

File: lib/web_ui/dev/driver_manager.dart [hunk 4]
```diff
@@ -137,11 +194,13 @@ abstract class DriverManager {
...
       return ChromeDriverManager(browser);
+    } else if (browser == 'firefox') {
+      return FirefoxDriverManager(browser);
     } else if (browser == 'safari' && io.Platform.isMacOS) {
       return SafariDriverManager(browser);
     } else {
-      throw StateError('Integration tests are only supported on Chrome or '
-          'on Safari (running on MacOS)');
+      throw StateError('Integration tests are only supported on Firefox, Chrome'
+          ' or on Safari (running on MacOS)');
```
Comment (yjbanov) [inline lib/web_ui/dev/driver_manager.dart:203]: nit: s/or/and/
nit: s/MacOS/macOS/ (throughout the PR)

File: lib/web_ui/dev/driver_manager.dart (hunk 3 above)
Comment (nturgut) [reply lib/web_ui/dev/driver_manager.dart:100]: This method is overriding the abstract class' method. I added `@override` everywhere to make it more clear.

In this class we always return `return Future<void>.value();` value. For example line 143.

**Comment by nturgut** [comment]
Thanks for the review!

File: lib/web_ui/dev/driver_manager.dart (hunk 2 above)
Comment (nturgut) [reply lib/web_ui/dev/driver_manager.dart]: done

File: lib/web_ui/dev/driver_manager.dart (hunk 1 above)
Comment (nturgut) [reply lib/web_ui/dev/driver_manager.dart]: :) done.

File: lib/web_ui/dev/driver_manager.dart (hunk 4 above)
Comment (nturgut) [reply lib/web_ui/dev/driver_manager.dart]: done

**Pull request merged**